tests/test_cli/test_issue_certificates.py

```diff
@@ -101,20 +95,26 @@ def setup_class(cls):
             [cls.cert_request_1, cls.cert_request_2], DummyConnection.connection_id.name
         )
 
-        # add fetchai key and connection key
-        shutil.copy(
-            FETCHAI_PRIVATE_KEY_PATH,
-            os.path.join(cls.current_agent_context, FETCHAI_PRIVATE_KEY_FILE),
-        )
-        cls.add_private_key()
-        cls.add_private_key(connection=True)
-
-    def test_issue_certificate(self):
+    def test_issue_certificate(self, password_or_none):
         """Test 'aea issue-certificates' in case of success."""
-        result = self.run_cli_command("issue-certificates", cwd=self._get_cwd())
+        # setup: add private key with password
+        self.generate_private_key(password=password_or_none)
+        self.add_private_key(password=password_or_none)
+        self.add_private_key(connection=True, password=password_or_none)
+
+        # issue certificates and check
+        password_options = _get_password_option_args(password_or_none)
+        result = self.run_cli_command(
+            "issue-certificates", *password_options, cwd=self._get_cwd()
+        )
         self._check_signature(self.cert_id_1, self.expected_path_1, result.stdout)
         self._check_signature(self.cert_id_2, self.expected_path_2, result.stdout)
 
+        # teardown: remove private key
+        Path(self._get_cwd(), FETCHAI_PRIVATE_KEY_FILE).unlink()
+        self.remove_private_key()
+        self.remove_private_key(connection=True)
```

Comment:
why not extract into setup and teardown methods?

Reply:
because `password_or_none` is a parametrized fixture, and the second run (either with password `None` or a mock password) would raise an error. The setup_class/teardown_class are run only one instead.

Reply:
setup and teardown are run between each test method call, setup_class/teardown_class are run at beginning and end...

Reply:
I know

Reply:
I'm sorry, I think I have not been clear enough on this. Since the setup part needs the password, I can't use the setup method because it does not support parametrization.